src/main/webapp/dashboard.test.js

```diff
@@ -0,0 +1,132 @@
+import * as dashboard from './dashboard.js';
+import '@testing-library/jest-dom/extend-expect';
+import {screen} from '@testing-library/dom';
+
+/**
+ * @fileOverview this file is essentially testing that if correct information
+ * about a user is gotten from the UserSevlet, It will be displayed in some way
+ * on the Dashboard.html page.
+ */
+global.mailtouiApp = {run: () => {}};
+
+const mockUserAllFields = {
+  interestTags: ['interest1', 'interest2'],
+  projects: [
+    {repositoryId: 'mockRepoID1', name: 'mockProjectName1'},
+    {repositoryId: 'mockRepoID2', name: 'mockProjectName2'},
+  ],
+  bio: 'This is the mock bio.',
+  email: 'samialves@google.com',
+  gitHubID: 'mockGitID',
+  gitHubURL: 'http://localhost/mockGitID',
+  image: 'http://localhost/image.jpg',
+  location: 'Mockville, Mocktana',
+  name: 'Mocky the Mockster',
+};
+
+const mockUserMissingOptionalFields = {
+  name: 'Mocky the Mockster',
+  image: 'http://localhost/image.jpg',
+  gitHubID: 'mockGitID',
+  gitHubURL: 'http://localhost/mockGitID',
+};
+
+// Note: Update this when the dashboard gets changed
+/**
+ * Function to replicate the DOM of Dashboard.html
+ */
+function resetDocumentBody() {
```

Comment:
Q: Is there a way for you to get this block from your html file?

Reply:
I looked into it and I couldn't figure out a good way. [this](https://dev.to/ms314006/use-jest-write-unit-testing-for-dom-manipulation-3n6c) was the resource I ended up using to decide how to write my tests. I'm very open to changing how the tests work, this feels very sweaty to me as it is.